codetf.schema-v3.json

```diff
@@ -0,0 +1,233 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://pixee.ai/codetf-v3.schema.json",
+  "title": "Code Transformation Format (CodeTF) v3.0",
+  "description": "A format for representing fixes, their associated changesets, and related metadata",
+  "type": "object",
+
+  "properties": {
+    "run": {
+      "description": "Metadata about the run that produced the results",
+      "$ref": "#/definitions/run"
+    },
+    "fixes": {
+      "description": "A list of fixes applied during the analysis",
+      "type": "array",
+      "items": { "$ref": "#/definitions/fix" },
+      "minItems": 0
+    },
+    "failedFiles": {
+      "description": "Files that the tool failed to process or transform",
+      "type": "array",
+      "items": { "$ref": "#/definitions/failedFile" },
+      "minItems": 0
+    },
+    "unfixedFindings": {
+      "description": "Findings that were detected but could not be fixed",
+      "type": "array",
+      "items": { "$ref": "#/definitions/unfixedFinding" },
+      "minItems": 0
+    }
+  },
+
+  "definitions": {
+    "run": {
+      "type": "object",
+      "properties": {
+        "vendor": { "type": "string", "description": "Name of the organization that produced the tool" },
+        "tool": { "type": "string", "description": "Name of the tool that produced the CodeTF results" },
+        "version": { "type": "string", "description": "Version of the tool that produced the CodeTF results" },
+        "projectName": { "type": "string", "description": "Name of the project that was analyzed" },
+        "elapsed": { "type": ["integer", "string"], "description": "Analysis duration in milliseconds" },
+        "commandLine": { "type": "string", "description": "Command line used to re-create this run (if applicable)" }
+      },
+      "required": ["vendor", "tool", "version", "elapsed"]
+    },
+
+    "fix": {
+      "type": "object",
+      "properties": {
+        "findingMetadata": {
+          "type": "object",
+          "description": "Metadata about the finding being addressed",
+          "properties": {
```

Comment:
I think it's worth pointing out all this `findingMetadata` should be from the tool itself -- which begs the question -- in the scope of a single issue being fixed, the caller already knows this -- so, do we need to send it back? Don't they already know it?